url2698a2c731_cassandra_project_cassandra_stand_alone_tgz-pJ8-eu_cassandra_sim_MySimpleDynamicScenarioSimulationJ8/src/SetupFileParser.java

```diff
@@ -0,0 +1,286 @@
+		propRequired.put("ScalarEnergyPricingTimeZones", requiredPropertiesPricingSEPTZ);
+		
+		String[] requiredPropertiesPricingEPP = { "contractedCapacity", "energyPrice", "powerPrice"};
+		propRequired.put("EnergyPowerPricing", requiredPropertiesPricingEPP);
+
+		String[] requiredPropertiesPricingMPP = { "energyPrice", "powerPrice"};
+		propRequired.put("MaximumPowerPricing", requiredPropertiesPricingMPP);
+		
+		String[] requiredPropertiesPricingAIP = { "fixedCost", "additionalCost", "contractedEnergy"};
+		propRequired.put("AllInclusivePricing", requiredPropertiesPricingAIP);
+		
+		String[] requiredPropertiesInstallation = { "id", "name"};
+		propRequired.put("installation", requiredPropertiesInstallation);
+		
+		String[] requiredPropertiesPerson = {"id", "name", "installation"};
+		propRequired.put("person", requiredPropertiesPerson);
+		
+		String[] requiredPropertiesConsModel = {"id", "name", "pmodel", "qmodel"};
+		propRequired.put("consumption_model", requiredPropertiesConsModel);
+		
+		String[] requiredPropertiesAppliance = {"id", "name", "installation", "consumption_model"};
+		propRequired.put("appliance", requiredPropertiesAppliance);
+		
+		String[] requiredPropertiesActivity = {"name", "person", "id"};
+		propRequired.put("activity", requiredPropertiesActivity);
+		
+		String[] requiredPropertiesActModel = {"id", "name", "activity", "containsAppliances", "day_type", "duration_distrType", "start_distrType", "repetitions_distrType"};
+		propRequired.put("activity_model", requiredPropertiesActModel);
+		
+	}
+	
+	public void parseFileForProperties(String filename)
+	{
+		try 
+		{
+			BufferedReader br = new BufferedReader(new FileReader(filename));
+			String line;
+			String segment = "";
+			String segmentTitle = "";
+			while ((line = br.readLine()) != null) {
+				if (line.startsWith("-->"))
+				{
+					if (!segment.equals(""))
+					{
+						parseSegment(segmentTitle, segment);
+						segment = "";
+					}
+					segmentTitle = line.replace("-->", "").trim();
+				}
+				else
+					segment += line + "\n";
+			}
+
+			if (!segment.equals(""))
+			{
+				parseSegment(segmentTitle, segment);
+			}
+
+			br.close();
+			
+			if (this.propSimulation.isEmpty())
+				throw new Exception("ERROR: required \"simulation\" segment missing from the setup file");
+			
+			if (this.propScenario.isEmpty())
+				throw new Exception("ERROR: required \"scenario\" segment missing from the setup file");
+			
+			if (this.propInstallations.isEmpty())
+				throw new Exception("ERROR: at least one \"installation\" segment must be present in the setup file");
+			
+			checkResponseType(propSimulation.getProperty("response_type"), propPricingBaseline, propPricing);
+			
+			
+		
+		} catch (IOException ex) {
+			ex.printStackTrace();
+		}
+		catch (Exception ex) {
+			System.err.println(ex.getMessage());
+			System.exit(0);
+		}
+	}
+	
+	private void parseSegment(String segmentTitle, String segment) throws Exception
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 14. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.